samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -1,276 +1,283 @@
+        for action in action_list:
+            new_statement += self._indent_string(f"   # - {action}\n", 7)
+
+        # APIGW API IAM Statements:
+        # https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-control-access-using-iam-policies-to-invoke-api.html
+        if resource_type in (ResourceType.APIGW_API, ResourceType.APIGW_REST_API):
+            apiId = self._extract_api_id_from_arn(resource_arn)
+            execute_api_arn = f"!Sub arn:${{AWS::Partition}}:execute-api:${{AWS::Region}}:${{AWS::AccountId}}:{apiId}/*"
+            return new_statement + self._indent_string(f"Resource: {execute_api_arn}\n", 7)
+
+        return new_statement + self._indent_string(f"Resource: {resource_arn}\n", 7)
+
+    def _get_default_bucket_name(self) -> str:
+        """
+        Returns a guaranteed unique default S3 Bucket name
+        """
+        return "test-runner-bucket-" + str(uuid.uuid4())
+
+    def generate_test_runner_template_string(self, image_uri: str) -> Union[dict, None]:
+        """
+        Creates a CloudFormation (YAML) Template string to create the Test Runner Stack.
+
+        Parameters
+        ---------
+
+        image_uri : str
+            The URI of the Image to be used by the Test Runner Fargate task definition.
+
+        Returns
+        -------
+        dict
+            The Test Runner CloudFormation template in the form of a YAML string.
+
+        Raises
+        ------
+        TestRunnerTemplateGenerationException
+            If the template generation process fails
+        """
+
+        default_bucket_name = self._get_default_bucket_name()
+
+        if self.resource_arn_list:
+            statements_list = []
+            for arn in self.resource_arn_list:
+                statement = self._create_iam_statement_string(arn)
+                if statement:
+                    statements_list.append(statement)
+            # Compile all the statements into a single string
+            generated_statements = "".join(statements_list)
+            if generated_statements:
+                return (
+                    self.TEST_RUNNER_TEMPLATE_STRING.replace("{{generated_statements}}", generated_statements)
+                    .replace("{{image_uri}}", image_uri)
+                    .replace("{{s3_bucket_name}}", default_bucket_name)
                 )
-            else:
-                return jinja2.Template(source=new_statement_template, keep_trailing_newline=True).render(
-                    action_list=action_list, arn=resource_arn
-                )
-
-    # If the supplied ARN is for a resource without any default actions supported
-    LOG.info("Found no match for ARN `%s` to any IAM actions.", resource_arn)
-    return jinja2.Template(source=new_statement_template, keep_trailing_newline=True).render(
-        action_list=[], arn=resource_arn
-    )
-
-
-def _query_tagging_api(tag_filters: dict, boto_client_provider: BotoProviderType) -> Union[dict, None]:
```

Comment:
Should I expect to see these again in a future PR?

Reply:
Yep, but it's not relevant to this one